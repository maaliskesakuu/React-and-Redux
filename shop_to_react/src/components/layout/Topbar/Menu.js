import React, { Component } from "react";

class Menu extends Component {
  render() {
    return (
      <>
        <div className="col-lg-6 text-center text-lg-right">
          <ul className="menu list-inline mb-0">
            <li className="list-inline-item">
              <a href="#" data-toggle="modal" data-target="#login-modal">
                Login
              </a>
            </li>
            <li className="list-inline-item">
              <a href="register.html">Register</a>
            </li>
            <li className="list-inline-item">
              <a href="contact.html">Contact</a>
            </li>
            <li className="list-inline-item">
              <a href="#">Recently viewed</a>
            </li>
          </ul>
        </div>
      </>
    );
  }
}

export default Menu;
