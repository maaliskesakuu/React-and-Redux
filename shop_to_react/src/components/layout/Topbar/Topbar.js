import React, { Component } from "react";
import LogIn from "../../auth/LogIn";
import Menu from "./Menu";

class Topbar extends Component {
  render() {
    return (
      <div id="top">
        <div className="container">
          <div className="row">
            <div className="col-lg-6 offer mb-3 mb-lg-0">
              <a href="#" className="btn btn-success btn-sm">
                Offer of the day
              </a>
              <a href="#" className="ml-1">
                Get flat 35% off on orders over $50!
              </a>
            </div>
            <Menu />
          </div>
        </div>
        <LogIn />
      </div>
    );
  }
}

export default Topbar;
