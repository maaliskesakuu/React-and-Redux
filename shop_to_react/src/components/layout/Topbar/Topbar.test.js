import React from "react";
import renderer from "react-test-renderer";
import Topbar from "./Topbar";
import { BrowserRouter } from "react-router-dom";

it("", () => {
  // cont wrapper = shallow(<Header/>);
  // console.log(wrapper.debug());

  const tree = renderer
    .create(
      <BrowserRouter>
        <Topbar />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
