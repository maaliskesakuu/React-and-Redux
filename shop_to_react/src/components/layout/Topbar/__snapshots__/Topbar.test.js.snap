// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  id="top"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-6 offer mb-3 mb-lg-0"
      >
        <a
          className="btn btn-success btn-sm"
          href="#"
        >
          Offer of the day
        </a>
        <a
          className="ml-1"
          href="#"
        >
          Get flat 35% off on orders over $50!
        </a>
      </div>
      <div
        className="col-lg-6 text-center text-lg-right"
      >
        <ul
          className="menu list-inline mb-0"
        >
          <li
            className="list-inline-item"
          >
            <a
              data-target="#login-modal"
              data-toggle="modal"
              href="#"
            >
              Login
            </a>
          </li>
          <li
            className="list-inline-item"
          >
            <a
              href="register.html"
            >
              Register
            </a>
          </li>
          <li
            className="list-inline-item"
          >
            <a
              href="contact.html"
            >
              Contact
            </a>
          </li>
          <li
            className="list-inline-item"
          >
            <a
              href="#"
            >
              Recently viewed
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="Login"
    className="modal fade"
    id="login-modal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog modal-sm"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Customer login
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body bg-dark"
      >
        <form
          action="customer-orders.html"
          method="post"
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="email-modal"
              placeholder="email"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="password-modal"
              placeholder="password"
              type="password"
            />
          </div>
          <p
            className="text-center"
          >
            <button
              className="btn btn-primary"
            >
              <i
                className="fa fa-sign-in"
              />
               Log in
            </button>
          </p>
        </form>
        <p
          className="text-center text-muted"
        >
          Not registered yet?
        </p>
        <p
          className="text-center text-muted"
        >
          <a
            href="register.html"
          >
            <strong>
              Register now
            </strong>
          </a>
          ! It is easy and done in 1 minute and gives you access to special discounts and much more!
        </p>
      </div>
    </div>
  </div>
</div>
`;
